usermods/PIR_sensor_switch/usermod_PIR_sensor_switch.h

```diff
@@ -283,21 +282,21 @@ void PIRsensorSwitch::publishHomeAssistantAutodiscovery()
 #ifndef WLED_DISABLE_MQTT
   if (WLED_MQTT_CONNECTED) {
     StaticJsonDocument<600> doc;
-    char uid[24], json_str[1024], buf[128];
+    char uid[128], json_str[1024], buf[256];
 
-    sprintf_P(buf, PSTR("%s Motion"), serverDescription); //max length: 33 + 7 = 40
+    sprintf_P(buf, PSTR("%s Motion %s "), serverDescription, _name); //max length: 33 + 7 = 40
     doc[F("name")] = buf;
-    sprintf_P(buf, PSTR("%s/motion"), mqttDeviceTopic);   //max length: 33 + 7 = 40
+    sprintf_P(buf, PSTR("%s/motion_%s"), mqttDeviceTopic,_name);   //max length: 33 + 7 = 40
     doc[F("stat_t")] = buf;
     doc[F("pl_on")]  = "on";
     doc[F("pl_off")] = "off";
-    sprintf_P(uid, PSTR("%s_motion"), escapedMac.c_str());
+    sprintf_P(uid, PSTR("%s_%s_motion"), escapedMac.c_str(),_name);
     doc[F("uniq_id")] = uid;
     doc[F("dev_cla")] = F("motion");
     doc[F("exp_aft")] = 1800;
 
     JsonObject device = doc.createNestedObject(F("device")); // attach the sensor to the same device
-    device[F("name")] = serverDescription;
+    device[F("name")] = String(serverDescription) ;

```

Comment:
Why String and unnecessary fragment heap?

Reply:
My bad , I work all the time woth esp32 so I don't pay lot of attention to limitation of esp8266. But noted